lib/galaxy/tools/toolbox/watcher.py

```diff
@@ -131,11 +133,16 @@ def check(self):
                     # in rare cases `path` may be deleted between `os.path.exists` calls
                     # and reading the file from the filesystem. We do not want the watcher
                     # thread to die in these cases.
-                    try:
-                        del hashes[path]
-                        del paths[path]
-                    except KeyError:
-                        pass
+                    if path in drop_now:
+                        log.warning("'%s' could not be read, removing from watched files", path)
+                        try:
+                            del hashes[path]
+                            del paths[path]
+                        except KeyError:
+                            pass
```

Comment:
If `path` is not in `hashes`, `path` would never be removed from `paths`.

```suggestion
                        del paths[path]
                        if path in hashes:
                            del hashes[path]
```